core/src/com/unciv/models/ruleset/nation/Personality.kt

```diff
@@ -0,0 +1,91 @@
+package com.unciv.models.ruleset.nation
+
+class Personality {
+
+    var diplomacy:HashMap<String, Float>
+    //Those two are not yet implemented
+    var buildings:HashMap<String, Float>
+    var citystates:HashMap<String, Float>
+
+    constructor(){
+        diplomacy = baseDiplomacy()
+        buildings = baseBuildings()
+        citystates = baseCityStates()
+    }
+    private fun baseDiplomacy():HashMap<String, Float>{
+        return hashMapOf(
+            "war" to 1f,
+            "trade" to 1f,
+            "openborders" to 1f,
+            "friendship" to 1f,
+            "alliance" to 1f
+        )
+    }
+
+    private fun baseBuildings():HashMap<String, Float>{
+        return hashMapOf(
+            "science" to 1f,
+            "production" to 1f,
+            "gold" to 1f,
+            "culture" to 1f,
+            "faith" to 1f,
+            "food" to 1f,
+            "happiness" to 1f,
+            "training" to 1f,
+            "military" to 1f,
+            "wonder" to 1f,
+            "worker" to 1f,
+            "workboat" to 1f
+        )
+    }
+
+    private fun baseCityStates():HashMap<String, Float>{
+        return hashMapOf(
+            "protect" to 1f,
+            "conquer" to 1f,
+            "ally" to 1f,
+            "bully" to 1f,
+            "interact" to 1f
```

Comment:
I don't like these magic strings. Each of these sections has *specific* values that are actually meaningful, a set of specific strings is best encapsulated in an Enum.

Even worse - you *require* these! e.g. below, diplomacy["war"]!! will crash the game if the modder forgot to add a war value!

What you actually need here, then, is not a hashmap at all.
It's classes. DiplomacyPersonality, ConstructionPersonality, CityStateInteractionPersonality, with *fields* for each of these values, with defaults. That way, if someone wants to only override the 'bully' personality (whatever that means) then he just needs to create a json with 1 value, rather than everything, and mods are future-safe as well, allowing us to add new values without ruining older mods.

Reply:
OK, will do it.